lightcycle-integration-test/src/test/java/com/soundcloud/lightcycle/integration_test/FragmentLoggerTest.java

```diff
@@ -1,132 +1,146 @@
 package com.soundcloud.lightcycle.integration_test;
 
-import static com.google.common.truth.Truth.assertThat;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.fragment.app.testing.FragmentScenario;
+import androidx.lifecycle.Lifecycle;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
 
 import com.google.common.truth.BooleanSubject;
 import com.soundcloud.lightcycle.integration_test.callback.FragmentLifecycleCallback;
-import com.soundcloud.lightcycle.sample.basic.BuildConfig;
+
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.robolectric.RobolectricTestRunner;
-import org.robolectric.android.controller.FragmentController;
-import org.robolectric.annotation.Config;
-import utils.FragmentTestHelper;
-
-import android.os.Bundle;
 
 import java.util.Arrays;
 import java.util.List;
 
-@RunWith(RobolectricTestRunner.class)
-@Config(constants = BuildConfig.class, sdk = 27)
+import utils.FragmentTestHelper;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(AndroidJUnit4.class)
```

Comment:
do we need this? is this deprecated?

Reply:
IIRC it uses Robolectric under the hood.

Reply:
nope.

Reply:
but anyways, nvm, this one isn't deprecated.

Reply:
> no, it doesn't. robolectric is treated like a third party library.

Alright, that's what I understood from http://robolectric.org/androidx_test/ but... whatever.

Reply:
woah you're right!

```
 private static String getRunnerClassName() {
    String runnerClassName = System.getProperty("android.junit.runner", null);
    if (runnerClassName == null) {
      // TODO: remove this logic when nitrogen is hooked up to always pass this property
      if (System.getProperty("java.runtime.name").toLowerCase().contains("android")) {
        return "androidx.test.internal.runner.junit4.AndroidJUnit4ClassRunner";
      } else {
        return "org.robolectric.RobolectricTestRunner";
      }
    }
    return runnerClassName;
  }
```

i woonder how that works if you haven't added the `RobolectricTestRunner` as a dependency.

Reply:
ALso, that's crazy! That has serious performance implications.

Reply:
this one is not deprecated, I updated to the latest one 
`android_x_test        : 'androidx.test.ext:junit:1.1.1'`